### PR TITLE
fix(Markdown): Don't wrap MarkdownAnchor component in tag overrides

### PR DESCRIPTION
**Pull request by casey-speer**
### Overview

Wrapping the `MarkdownAnchor` in an anonymous functional component in the `Markdown` component's default 'a' tag override causes react to crash when certain plugins (like google translate) modify the dom and react tries to reconcile the changes.

Removing the wrapper fixes the issue, though I'm still not certain why this distinction matters. Welcome any thoughts about why this might be the case and if a different fix would be appropriate.

See the JIRA ticket for additional context

To test:
1. Install the alpha `gamut` package in a local copy of the monolith
2. Visit localhost:3000/courses/introduction-to-javascript/lessons/introduction-to-javascript/exercises/built-in-methods or any other lesson with an 'a' tag in the narrative content
3. Use google translate to change the page to a different language (see the JIRA ticket for more details on how to do this)
4. Navigate through the narrative and code editor and check that interacting with various elements does not cause react to blow up 

Fixes react crashing due to wrapping the `'a'` tag Markdown override, `MarkdownAnchor`, in a functional component.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1214
- [x] I have run this code to verify it works
- [x] ~This PR includes unit tests for the code change~ (This is a very rare edge case and hard to check with units tests)

### CODE REVIEW

File: packages/gamut/src/Markdown/index.tsx [hunk 1]
```diff
@@ -83,9 +84,10 @@ export class Markdown extends PureComponent<MarkdownProps> {
         }),
       !skipDefaultOverrides.a &&
         createTagOverride('a', {
-          component: (props: MarkdownAnchorProps) => (
-            <MarkdownAnchor onClick={onAnchorClick} {...props} />
-          ),
+          component: MarkdownAnchor,
+          processNode: (node, props) =>  {
```
Comment (casey-speer) [inline packages/gamut/src/Markdown/index.tsx:88]: Open to thoughts on different better solutions here.

File: packages/gamut/src/Markdown/index.tsx [hunk 2]
```diff
@@ -83,9 +84,10 @@ export class Markdown extends PureComponent<MarkdownProps> {
...
+            return <MarkdownAnchor onClick={onAnchorClick} {...props as MarkdownAnchorProps} />;
```
Comment (casey-speer) [inline packages/gamut/src/Markdown/index.tsx:89]: @JoshuaKGoldberg is there a better way to express the `props` type here?

File: packages/gamut/src/Markdown/index.tsx (hunk 1 above)
Comment (jakemhiller) [reply packages/gamut/src/Markdown/index.tsx:88]: I wonder if the function override had issues because it referenced a value defined in the parent component? Maybe there's some goofy rerender behavior where it loses that scope? Either way maybe we should just switch to using `processNode` by default 🤷🏻‍♂️

**Review by jakemhiller — APPROVED** [review]
Lgtm! I bet this was the issue that caused some other bugs w/ the anchor override

**Review by JoshuaKGoldberg — APPROVED** [review]
🤷 🙃 . Nice investigation & find!

File: packages/gamut/src/Markdown/index.tsx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Markdown/index.tsx:89]: You'd want to add a generic to `OverrideSettings`, I think, that defaults to `{}` or `object` or some equivalent. Right now `OverrideSettings` only knows `props: object` to be the case, always, no matter what.

```diff
- export type OverrideSettings = {
+ export type OverrideSettings<Props = {}> = {
...
- props: object
+ props: Props
```

...I'm not positive this is it, but you could try playing around. I feel like something similar to this should work. Then you'd be able to say:

```ts
processNode: (node, props: MarkdownAnchorProps) => {
```

File: packages/gamut/src/Markdown/index.tsx [hunk 3]
```diff
@@ -83,9 +84,10 @@ export class Markdown extends PureComponent<MarkdownProps> {
         }),
       !skipDefaultOverrides.a &&
         createTagOverride('a', {
-          component: (props: MarkdownAnchorProps) => (
-            <MarkdownAnchor onClick={onAnchorClick} {...props} />
-          ),
+          component: MarkdownAnchor,
```
[the remark this answers is not in the document]
Comment (casey-speer) [reply packages/gamut/src/Markdown/index.tsx:87]: The types do require it. I was hesitant to make this optional but maybe I can make two separate types: one where this is required and one where `processNode` is required.

**Review by codecaaron — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.1-alpha.f76fd1.0<br/>@codecademy/gamut-labs@7.7.4-alpha.f76fd1.0<br/>@codecademy/gamut@19.9.4-alpha.f76fd1.0<br/>@codecademy/styleguide@21.9.2-alpha.f76fd1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60072636e8e973482cc8856f--gamut-preview.netlify.app>https://60072636e8e973482cc8856f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60072636e8e973482cc8856f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.5-alpha.921827.0<br/>@codecademy/gamut-labs@7.10.5-alpha.921827.0<br/>@codecademy/gamut@19.11.1-alpha.921827.0<br/>@codecademy/styleguide@21.15.1-alpha.921827.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60072deeced2c078412a8a43--gamut-preview.netlify.app>https://60072deeced2c078412a8a43--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60072deeced2c078412a8a43>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.5-alpha.f694d5.0<br/>@codecademy/gamut-labs@7.10.5-alpha.f694d5.0<br/>@codecademy/gamut@19.11.1-alpha.f694d5.0<br/>@codecademy/styleguide@21.15.2-alpha.f694d5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600733538cb5f068eb55ac45--gamut-preview.netlify.app>https://600733538cb5f068eb55ac45--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600733538cb5f068eb55ac45>Deploy Logs</a></p>
